ingestion/src/metadata/utils/dbt_config.py

```diff
@@ -0,0 +1,113 @@
+        logger.debug(f"Error fetching dbt files from gcs {repr(exc)}")
+    return None
+
+
+@get_dbt_details.register
+def _(config: DbtLocalConfig):
+    try:
+        if config.dbtCatalogFilePath is not None:
+            with open(config.dbtCatalogFilePath, "r", encoding="utf-8") as catalog:
+                dbt_catalog = catalog.read()
+        if config.dbtManifestFilePath is not None:
+            with open(config.dbtManifestFilePath, "r", encoding="utf-8") as manifest:
+                dbt_manifest = manifest.read()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from local {repr(exc)}")
+    return None
+
+
+@get_dbt_details.register
+def _(config: DbtHttpConfig):
+    try:
+        catalog_file = urllib.request.urlopen(config.dbtCatalogHttpPath)
+        manifest_file = urllib.request.urlopen(config.dbtManifestHttpPath)
+        dbt_catalog = catalog_file.read().decode()
+        dbt_manifest = manifest_file.read().decode()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from file server {repr(exc)}")
+    return None
+
+
+@get_dbt_details.register
+def _(config: DbtS3Config):
+    try:
+        from metadata.utils.aws_client import AWSClient
+
+        aws_client = AWSClient(config.dbtSecurityConfig).get_resource("s3")
+        buckets = aws_client.buckets.all()
+        for bucket in buckets:
+            for bucket_object in bucket.objects.all():
+                if DBT_MANIFEST_FILE_NAME in bucket_object.key:
+                    dbt_manifest = bucket_object.get()["Body"].read().decode()
+                if DBT_CATALOG_FILE_NAME in bucket_object.key:
+                    dbt_catalog = bucket_object.get()["Body"].read().decode()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from s3 {repr(exc)}")
+    return None
+
+
+@get_dbt_details.register
+def _(config: DbtGCSConfig):
+    try:
+        from google.cloud import storage
+
+        set_google_credentials(gcs_credentials=config.dbtSecurityConfig)
+        client = storage.Client()
+        for bucket in client.list_buckets():
+            for blob in client.list_blobs(bucket.name):
+                if DBT_MANIFEST_FILE_NAME in blob.name:
+                    dbt_manifest = blob.download_as_string().decode()
+                if DBT_CATALOG_FILE_NAME in blob.name:
+                    dbt_catalog = blob.download_as_string().decode()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from gcs {repr(exc)}")
```

Comment:
even if we return none, let's log.error this

Reply:
same for the others, if we cannot properly get the files we should show that clearly

Reply:
Yes